nx/include/switch/kernel/semaphore.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Kevoot
+
+#ifndef __SEMAPHORE_H
+#define __SEMAPHORE_H
+
+#include "mutex.h"
+
+#define SEM_DOWN false
+#define SEM_UP   true
+
+#define EBUSY 1
```

Comment:
Do not redefine this, I don't see why you need to use EBUSY, it's not used anywhere.